### PR TITLE
Integration tests for Preferential GPU Scheduling in Marathon

### PR DESCRIPTION
**Pull request by meln1k**
Summary: Integration test for checking the correctness of the gpu behavior implementation

JIRA issues: MARATHON-8092

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      Then("there should be a match regardless selected policy")
+      waitForStatusUpdates("TASK_RUNNING")
+      waitForDeployment(update)
+      waitForStatusUpdates("TASK_FINISHED")
+    }
+
+    "reject offers for new apps with a persistent volume and no GPUs" in {
+      Given("An app that writes into a persistent volume")
+      val containerPath = "persistent-volume"
+      val applicationId = appId(Some("new-resident-task-no-matach"))
```
Comment (ichernetsky) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: A typo: `no-matach`.

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      When("the app is suspended")
+      suspendSuccessfully(PathId(app.id))
+
+      And("a new task is started that checks for the previously written file")
+      // deploy a new version that checks for the data written the above step
+      val update = marathon.updateApp(
+        PathId(app.id),
+        AppUpdate(
+          instances = Some(1),
+          cmd = Some(s"""test -e $containerPath/data && sleep 2""")
```
Comment (ichernetsky) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: Doesn't this `cmd` always returns `0`?

Comment (meln1k) [reply src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: Why do you think so?

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      val app = appProxy(appId(Some("with-gpu-resources")), "v1", instances = 1, healthCheck = None, gpus = 1)
+
+      When("The app is deployed")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created")
+      result should be(Created)
+      extractDeploymentIds(result) should have size 1
+      waitForDeployment(result)
+      waitForTasks(app.id.toPath, 1) //make sure, the app has really started
```
Comment (alenkacz) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: space after // :)

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    "not match any offers an app without GPU requirements" in {
```
Comment (alenkacz) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: not match any offers *for* ... ? I feel like there is some word missing...

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+      When("The app is deployed")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created")
+      result should be(Created)
+      extractDeploymentIds(result) should have size 1
+
+      And("DeclinedScarceResources reject must happen")
+      waitForAppOfferReject(applicationId, "DeclinedScarceResources")
+
```
Comment (alenkacz) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: extra space

File: src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      portDefinitions = Some(portDefinitions),
+      backoffSeconds = backoffDuration.toSeconds.toInt,
+      upgradeStrategy = Some(UpgradeStrategy(minimumHealthCapacity = 0.5, maximumOverCapacity = 0.0)),
+      unreachableStrategy = Some(UnreachableDisabled.DefaultValue)
+    )
+
+    app
+  }
+
+  def createSuccessfully(app: App): App = {
```
Comment (alenkacz) [inline src/test/scala/mesosphere/marathon/integration/GpuSchedulingIntegrationTest.scala]: this does not seem to be used ... so maybe remove it?

**Pull request merged**